### PR TITLE
feat: Complete Printer Profiles API implementation

### PR DESCRIPTION
**Pull request by sophiedeziel**
## Summary
Complete implementation of the OctoPrint Printer Profiles API with full CRUD operations and comprehensive test coverage.

## Features Implemented
- **List all printer profiles** (`GET /api/printerprofiles`)
- **Get specific printer profile** (`GET /api/printerprofiles/{identifier}`)  
- **Create new printer profile** (`POST /api/printerprofiles`)
  - Support for creating profiles based on existing profiles (`basedOn` parameter)
- **Update existing printer profile** (`PATCH /api/printerprofiles/{identifier}`)
- **Delete printer profile** (`DELETE /api/printerprofiles/{identifier}`)

## Technical Implementation
- **Full Sorbet type coverage** with proper type signatures
- **Zeitwerk autoloading** compatible structure
- **AutoInitializable and Deserializable mixins** for consistent object creation
- **Proper API field mapping** (camelCase ↔ snake_case conversion)
- **Comprehensive error handling** with proper exception types

## Test Coverage
- **14 unit tests** covering all class methods and edge cases
- **45 integration tests** with real API interactions via VCR cassettes  
- **7 VCR cassettes** covering all API endpoints and scenarios:
  - `list.yml` - List all profiles
  - `get_default.yml` - Get specific profile
  - `create.yml` - Create new profile
  - `create_based_on.yml` - Create profile based on existing
  - `update.yml` - Update existing profile  
  - `delete.yml` - Delete profile
  - `delete_verification.yml` - Verify deletion
- **Additional camelCase conversion tests** for Deserializable module

## Code Quality
- ✅ **RuboCop**: All linting rules pass
- ✅ **Sorbet**: Full type checking passes  
- ✅ **Test Coverage**: **100.0% line coverage (622/622 lines)**
- ✅ **No curl dependencies**: Tests use VCR cassettes exclusively

## Recent Updates
- Added comprehensive tests for camelCase to snake_case conversion in Deserializable module
- Restored test coverage to 100% by covering previously untested automatic key conversion functionality
- Enhanced test suite with 3 additional test cases covering edge cases in key transformation

## API Documentation Reference
Implementation follows [OctoPrint Printer Profiles API v1.11.2](https://docs.octoprint.org/en/1.11.2/api/printerprofiles.html)

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: lib/octoprint/deserializable.rb [hunk 1]
```diff
@@ -409,23 +487,13 @@ def transform(&block)
...
-        @handle_extras = true
-      end
-
-      # @return [Boolean] Whether extra field collection is enabled
-      # @api private
-      sig { returns(T::Boolean) }
-      def handle_extras?
-        @handle_extras
+        # No-op - extras collection is now automatic
       end
```
Comment (sophiedeziel) [inline lib/octoprint/deserializable.rb:497]: You can delete that method
